pyaedt/icepak.py

```diff
@@ -4869,3 +4869,182 @@ def assign_symmetry_wall(
+        >>> box.solve_inside = False
+        >>> recirc = ipk.assign_recirculation_opening([box.top_face_x, box.bottom_face_x], box.top_face_x,
+        >>>                                          flow_assignment="10kg_per_s_m2")
+
+        """
+        if not len(face_list) == 2:
+            self.logger.error("Recirculation boundary condition must be assigned to two faces.")
+            return False
+        if conductance_external_temperature is not None and thermal_specification is not "Conductance":
+            self.logger.warning(
+                '``conductance_external_temperature`` will not have any effect unless the ``thermal_specification`` '
+                'is ``"Conductance"``.')
+        if conductance_external_temperature is not None and thermal_specification is not "Conductance":
+            self.logger.warning(
+                '``conductance_external_temperature`` needs to be specified when ``thermal_specification`` '
+                'is ``"Conductance"``. Setting ``conductance_external_temperature`` to ``"AmbientTemp"``.')
+        if (start_time is not None or end_time is not None) and not self.solution_type == "Transient":
+            self.logger.warning(
+                '``start_time`` and ``end_time`` will not have any effect unless for steady-state simulations.')
+        elif self.solution_type == "Transient" and not (start_time and end_time):
+            self.logger.warning(
+                '``start_time`` and ``end_time`` should be declared for transient simulations. Setting them to "0s".')
+            start_time = "0s"
+            end_time = "0s"
+        assignment_dict = {
+            "Conductance": "Conductance",
+            "Heat Input": "Heat Flow",
+            "Temperature": "Temperature Change"
+        }
+        props = {}
+        if not isinstance(face_list[0], int):
+            face_list = [f.id for f in face_list]
+        props["Faces"] = face_list
+        if isinstance(extract_face, int):
+            extract_face = [extract_face]
+        else:
+            extract_face = [extract_face.id]
+        props["ExtractFace"] = extract_face
+        props["Thermal Condition"] = thermal_specification
+        if isinstance(assignment_value, dict):
+            if not self.solution_type == "Transient":
+                self.logger.error("Transient assignment is supported only in transient designs.")
+                return None
+            assignment = self._parse_variation_data(
+                assignment_dict[thermal_specification],
+                "Transient",
+                variation_value=assignment_value["Values"],
+                function=assignment_value["Function"],
+            )
+            props.update(assignment)
+        else:
+            props[assignment_dict[thermal_specification]] = assignment_value
+        if thermal_specification == "Conductance":
+            props["External Temp"] = conductance_external_temperature
+        if isinstance(flow_assignment, dict):
```

Comment:
@lorenzovecchietti Could you add a unit test to cover this case?

Reply:
Done